llvm/lib/TableGen/TGLexer.cpp

```diff
@@ -51,14 +67,14 @@ static const char *lexMacroName(StringRef Str) {
 
   // Macro names start with [a-zA-Z_].
   const char *Next = Str.begin();
-  if (*Next != '_' && !isalpha(*Next))
+  if (!IsValidIDChar(*Next, First))
```

Comment:
Just use a bool.

First/NotFirst that pollute the namespace isn't better than a boolean here.

Reply:
Done.